setup.py

```diff
@@ -1,77 +1,174 @@
+
+
+    def _minimal_ext_cmd(cmd):
+        # construct minimal environment
+        env = {}
+        for k in ['SYSTEMROOT', 'PATH']:
+            v = os.environ.get(k)
+            if v is not None:
+                env[k] = v
+        # LANGUAGE is used on win32
+        env['LANGUAGE'] = 'C'
+        env['LANG'] = 'C'
+        env['LC_ALL'] = 'C'
+        out = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, env=env,
+        ).communicate()[0]
+        return out
+
+    try:
+        out = _minimal_ext_cmd(['git', 'describe', '--tags'])
+    except OSError:
+        out = ''
+
+    git_description = out.strip().decode('ascii')
+    expr = r'.*?\-(?P<count>\d+)-g(?P<hash>[a-fA-F0-9]+)'
+    match = re.match(expr, git_description)
+    if match is None:
+        git_revision, git_count = 'Unknown', '0'
+    else:
+        git_revision, git_count = match.group('hash'), match.group('count')
 
-info = {}
-execfile(join('chaco', '__init__.py'), info)
+    return git_revision, git_count
 
-numpy_include_dir = get_include()
 
+def write_version_py(filename=_VERSION_FILENAME):
+    template = """\
+# THIS FILE IS GENERATED FROM CHACO SETUP.PY
+version = '{version}'
+full_version = '{full_version}'
+git_revision = '{git_revision}'
+is_released = {is_released}
 
-# Register Python extensions
-contour = Extension(
-    'chaco.contour.contour',
-    sources=['chaco/contour/cntr.c'],
-    include_dirs=[numpy_include_dir],
-    define_macros=[('NUMPY', None)]
+if not is_released:
+    version = full_version
+"""
+    # Adding the git rev number needs to be done inside
+    # write_version_py(), otherwise the import of _version messes
+    # up the build under Python 3.
+    fullversion = VERSION
+    if os.path.exists('.git'):
+        git_rev, dev_num = git_version()
+    elif os.path.exists(filename):
+        # must be a source distribution, use existing version file
+        try:
+            from chaco._version import git_revision as git_rev
+            from chaco._version import full_version as full_v
+        except ImportError:
+            msg = ("Unable to import 'git_revision' or 'full_revision'. "
+                   "Try removing {} and the build directory before building.")
+            raise ImportError(msg.format(_VERSION_FILENAME))
+
+        match = re.match(r'.*?\.dev(?P<dev_num>\d+)', full_v)
+        if match is None:
+            dev_num = '0'
+        else:
+            dev_num = match.group('dev_num')
+    else:
+        git_rev = 'Unknown'
+        dev_num = '0'
+
+    if not IS_RELEASED:
+        fullversion += '.dev{0}'.format(dev_num)
+
+    with open(filename, "wt") as fp:
+        fp.write(template.format(version=VERSION,
+                                 full_version=fullversion,
+                                 git_revision=git_rev,
+                                 is_released=IS_RELEASED))
+
+
+if __name__ == "__main__":
+    write_version_py()
+    from chaco import __requires__, __version__
```

Comment:
This looks like an old style relative import. I think it will cause problems with python 3 in the future.